source/DataElement/AMDSBufferGroup.h

```diff
@@ -55,12 +55,15 @@ class AMDSBufferGroup : public QObject
 public slots:
 	/// Slot which handles a request for data. The buffer group will attempt to populate the request
 	/// based on the instructions it includes. When the data request is ready the dataRequestReady signal is emitted
-	void processClientRequest(AMDSClientRequest *clientRequest);
+	void processClientRequest(AMDSClientRequest *clientRequest, bool internalRequest = false);
 
 signals:
 	/// Signal which indicates that a request for data has been processed and is ready to be sent back to the client
 	void clientRequestProcessed(AMDSClientRequest *clientRequest);
 
+	/// Handles clientRequests that have been requested internally and don't need to be routed out of the server (flattening request for trigger/dwell)
+	void internalRequestProcessed(AMDSClientRequest *clientRequest);
```

Comment:
Is there a reason why you had to add the internalRequest and it couldn't be done with a general clientRequest?  I'm assuming you aren't wanting extra signals to send the wrong messages?